runtime/util/cphelp.c

```diff
@@ -149,16 +152,41 @@ getModuleJRTURL(J9VMThread *currentThread, J9ClassLoader *classLoader, J9Module
 	if (NULL == jrtURL) {
 		if (J9_ARE_ALL_BITS_SET(javaVM->runtimeFlags, J9_RUNTIME_JAVA_BASE_MODULE_CREATED)) {
 			/* set jrt URL for the module */
-			jrtURL = vmFuncs->copyStringToJ9UTF8WithMemAlloc(currentThread, module->moduleName, J9_STR_NONE, "jrt:/", 5, NULL, 0);
-
+			const char *prependStr = "jrt:/";
+			const size_t prependStrLen = strlen(prependStr);
+			const J9UTF8 *moduleName = module->moduleName;
+			const U_8 *nameData = J9UTF8_DATA(moduleName);
+			const U_16 nameLength = J9UTF8_LENGTH(moduleName);
+			const U_16 totalStringLength = prependStrLen + nameLength;
+			const UDATA jrtURLSize = totalStringLength + sizeof(J9UTF8);
+#if defined(J9VM_OPT_SNAPSHOTS)
+			if (IS_SNAPSHOTTING_ENABLED(javaVM)) {
+				jrtURL = (J9UTF8 *)vmsnapshot_allocate_memory(jrtURLSize + 1, OMRMEM_CATEGORY_VM);
+			} else
+#endif /* defined(J9VM_OPT_SNAPSHOTS) */
+			{
+				jrtURL = (J9UTF8 *)j9mem_allocate_memory(jrtURLSize + 1, OMRMEM_CATEGORY_VM);
+			}
 			if (NULL == jrtURL) {
 				goto _exit;
 			}
+			memcpy(J9UTF8_DATA(jrtURL), prependStr, prependStrLen);
+			memcpy(J9UTF8_DATA(jrtURL) + prependStrLen, nameData, nameLength);
+			J9UTF8_DATA(jrtURL)[totalStringLength] = '\0';
+			J9UTF8_SET_LENGTH(jrtURL, totalStringLength);
```

Comment:
Code to fix the merge conflict.
```suggestion
                        if (NULL == module->moduleName) {
                                goto _exit;
                        } else {
                        	/* Set jrt URL for the module. */
				const char *prependStr = "jrt:/";
				const size_t prependStrLen = strlen(prependStr);
				const J9UTF8 *moduleName = module->moduleName;
				const U_8 *nameData = J9UTF8_DATA(moduleName);
				const U_16 nameLength = J9UTF8_LENGTH(moduleName);
				const U_16 totalStringLength = prependStrLen + nameLength;
				const UDATA jrtURLSize = totalStringLength + sizeof(J9UTF8);
#if defined(J9VM_OPT_SNAPSHOTS)
				if (IS_SNAPSHOTTING_ENABLED(javaVM)) {
					jrtURL = (J9UTF8 *)vmsnapshot_allocate_memory(jrtURLSize + 1, OMRMEM_CATEGORY_VM);
				} else
#endif /* defined(J9VM_OPT_SNAPSHOTS) */
				{
					jrtURL = (J9UTF8 *)j9mem_allocate_memory(jrtURLSize + 1, OMRMEM_CATEGORY_VM);
				}
				if (NULL == jrtURL) {
					goto _exit;
				}
				memcpy(J9UTF8_DATA(jrtURL), prependStr, prependStrLen);
				memcpy(J9UTF8_DATA(jrtURL) + prependStrLen, nameData, nameLength);
				J9UTF8_DATA(jrtURL)[totalStringLength] = '\0';
				J9UTF8_SET_LENGTH(jrtURL, totalStringLength);
                        }
```

Reply:
I don't think we need the `else` on line 157, but thank you for the patch.